solr/core/src/java/org/apache/solr/query/SolrRangeQuery.java

```diff
@@ -521,39 +511,59 @@ private SegState getSegState(LeafReaderContext context) throws IOException {
       return segStates[context.ord] = new SegState(segSet);
     }
 
-    private Scorer scorer(DocIdSet set) throws IOException {
+    private Scorer scorerInternal(DocIdSet set) throws IOException {
       if (set == null) {
         return null;
       }
       final DocIdSetIterator disi = set.iterator();
       if (disi == null) {
         return null;
       }
-      return new ConstantScoreScorer(this, score(), scoreMode, disi);
+      return new ConstantScoreScorer(score(), scoreMode, disi);
     }
 
-    @Override
-    public BulkScorer bulkScorer(LeafReaderContext context) throws IOException {
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    public Scorer scorerInternal(LeafReaderContext context) throws IOException {
       final SegState weightOrBitSet = getSegState(context);
+      log.info("Query: " + getQuery());
+      log.info("weight: " + weightOrBitSet.weight);
+      if (weightOrBitSet.weight!=null) log.info("weight's scorer: " + weightOrBitSet.weight.scorer(context));
+      log.info("set: " + weightOrBitSet.set);
       if (weightOrBitSet.weight != null) {
-        return weightOrBitSet.weight.bulkScorer(context);
+        Scorer ret = weightOrBitSet.weight.scorer(context);
+        return ret;
       } else {
-        final Scorer scorer = scorer(weightOrBitSet.set);
-        if (scorer == null) {
-          return null;
-        }
-        return new DefaultBulkScorer(scorer);
+        return scorerInternal(weightOrBitSet.set);
       }
     }
 
     @Override
-    public Scorer scorer(LeafReaderContext context) throws IOException {
-      final SegState weightOrBitSet = getSegState(context);
-      if (weightOrBitSet.weight != null) {
-        return weightOrBitSet.weight.scorer(context);
-      } else {
-        return scorer(weightOrBitSet.set);
+    public ScorerSupplier scorerSupplier(LeafReaderContext context) throws IOException {
```

Comment:
[INFO] The "ScoreSupplier" interface and method was added to Lucene a while back as a way to do better query-planning for range queries, but https://github.com/apache/lucene/issues/13180 recently shifted the method to being "abstract" to force all Weight subclasses to implement it.

`scorer(..)` now has a default implementation that delegates to scoreSupplier, so in effect most Weight impl's have had to switch out their `scorer(...)` method for a `scorerSupplier(...)`.